rudder-lang/src/io.rs

```diff
@@ -32,187 +32,168 @@ struct Config {
 }
 
 /// IO context deduced from arguments (structopt) and config file (Config)
+// must always reflect GenericOptions + add unique fields
 #[derive(Clone, Debug)]
 pub struct IOContext {
+    // GenericOption reflection
     pub stdlib: PathBuf,
-    pub source: PathBuf,
-    pub dest: PathBuf,
-    pub mode: Action,
+    pub input: PathBuf,
+    pub output: PathBuf,
+    // Unique fields
     pub format: Format,
+    pub action: Action,
+}
+// TODO might try to merge io.rs in here
+impl IOContext {
+    pub fn set(action: Action, opt: &GenericOptions, format: Option<Format>) -> Result<Self> {
```

Comment:
set is a too generic name, what does it do ? 
isn't thei a new() ?

Reply:
yes, will be updated